src/pylhe/__init__.py

```diff
@@ -496,14 +496,19 @@ def read_lhe(filepath):
     """
     try:
         with _extract_fileobj(filepath) as fileobj:
-            for _event, element in ET.iterparse(fileobj, events=["end"]):
-                if element.tag == "event":
+            context = ET.iterparse(fileobj, events=["start", "end"])
+            _, root = next(context)  # Get the root element
+            for event, element in context:
+                if event == "end" and element.tag == "event":
                     data = element.text.strip().split("\n")
                     eventdata, particles = data[0], data[1:]
                     eventinfo = LHEEventInfo.fromstring(eventdata)
                     particles = particles[: int(eventinfo.nparticles)]
                     particle_objs = [LHEParticle.fromstring(p) for p in particles]
                     yield LHEEvent(eventinfo, particle_objs)
+                    # Clear the element to free memory
+                    element.clear()
+                    root.clear()
```

Comment:
Can you explain for my own understanding why `root.clear()` works fine *within* the loop? Maybe worth adding a comment?

Reply:
ahh, yes, `root` tracks all the sub-elements. https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.clear removes the references to the sub-elements, thus clearing it gives back memory.